File: user-service/src/main/java/io/github/kstnnn/user/service/converter/UserStatusConverter.java
package io.github.kstnnn.user.service.converter;

import io.github.kstnnn.user.service.model.UserStatus;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class UserStatusConverter implements AttributeConverter<UserStatus, String> {
  @Override
  public String convertToDatabaseColumn(UserStatus attribute) {
    return attribute != null ? attribute.name() : null;
  }

  @Override
  public UserStatus convertToEntityAttribute(String dbData) {
    return dbData != null ? UserStatus.valueOf(dbData) : null;
  }
}

File: user-service/src/main/kotlin/io/github/kstnnn/user/service/converter/UserStatusConverter.kt
package io.github.kstnnn.user.service.converter

import io.github.kstnnn.user.service.enums.UserStatus
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class UserStatusConverter : AttributeConverter<UserStatus, String> {
    override fun convertToDatabaseColumn(attribute: UserStatus?): String? = attribute?.name

    override fun convertToEntityAttribute(dbData: String?): UserStatus? = dbData?.let { UserStatus.valueOf(it) }
}
